sentry.client.config.ts

```diff
@@ -1,17 +1,59 @@
+import {
+  BrowserClient,
+  Dedupe,
+  Replay,
+  Breadcrumbs,
+  HttpContext,
+  BrowserTracing,
+  defaultStackParser,
+  getCurrentHub,
+  makeFetchTransport,
+} from '@sentry/nextjs';
+
 import { SENTRY_DSN, SENTRY_ENABLE } from '@/next.constants.mjs';
 
-// This lazy-loads Sentry on the Browser
-import('@sentry/nextjs').then(({ init }) =>
-  init({
-    // Only run Sentry on Vercel Environment
-    enabled: SENTRY_ENABLE,
-    // Tell Sentry where to send events
-    dsn: SENTRY_DSN,
-    // Disable Sentry Tracing as we don't need to have it
-    // as Vercel already does Web Vitals and Performance Measurement on Client-Side
-    enableTracing: false,
-    // We only want to capture errors from _next folder on production
-    // We don't want to capture errors from preview branches here
-    allowUrls: ['https://nodejs.org/_next'],
-  })
-);
+// This creates a custom Sentry Client with minimal integrations
+export const sentryClient = new BrowserClient({
+  // Only run Sentry on Vercel Environment
+  enabled: SENTRY_ENABLE,
+  // Provide Sentry's Secret Key
+  dsn: SENTRY_DSN,
+  // Sentry's Error Transport Mechanism
+  transport: makeFetchTransport,
+  // Sentry's Stack Trace Parser
+  stackParser: defaultStackParser,
+  // All supported Integrations by us
+  integrations: [
+    new Replay(),
+    new Dedupe(),
+    new HttpContext(),
+    new Breadcrumbs(),
+    new BrowserTracing(),
```

Comment:
You'll also want the `LinkedErrors` here as well, esp while using `withErrorBoundary`.

Youre also disabling https://docs.sentry.io/platforms/javascript/configuration/integrations/globalhandlers/ with this config, is that intended? You won't capture any errors that bubble up to global handlers.

Reply:
> Youre also disabling [docs.sentry.io/platforms/javascript/configuration/integrations/globalhandlers](https://docs.sentry.io/platforms/javascript/configuration/integrations/globalhandlers/) with this config, is that intended? You won't capture any errors that bubble up to global handlers.

Right, we want to ignore unhandled rejections for now.

> You'll also want the LinkedErrors here as well, esp while using withErrorBoundary.

Interesting... can you elaborate what LinkedErrors does?

Reply:
> Right, we want to ignore unhandled rejections for now.

You'll also lose out on errors that bubble up to `window.onerror` as well.

> can you elaborate what LinkedErrors does

[LinkedErrors](https://docs.sentry.io/platforms/javascript/configuration/integrations/linkederrors/) adds support for [`error.cause`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause), which can be used for chained errors.

This is used in the error boundary for linking component stacks to errors and getting everything sourcemapped: https://docs.sentry.io/platforms/javascript/guides/react/features/error-boundary/#linked-errors

> Also is withErrorBoundary even important to be used? As it's not part of default Next.js setup?

It's not part of the default Next.js setup because apps are all different, it's hard to tell where to add this for users since you also need to configure a fallback UI for it. I recommend using it, it's helpful to generate failures states that users can respond to/reset. https://docs.sentry.io/platforms/javascript/guides/react/features/error-boundary